tensilelite/Tensile/KernelWriter.py

```diff
@@ -4852,28 +4852,9 @@ def readWriteVectors(mat, vw, kernel):
       self.states.localReadDoCntMetadata  = 0
 
     if kernel["EnableMatrixInstruction"]:
-      numBytes = kernel["ProblemType"]["DataType"].numBytes()
-      mi_divisor = 2
-      miIssueLatency = 2
-      if (self.states.version == (9,4,0) or self.states.version == (9,4,1) or self.states.version == (9,4,2) or self.states.version == (9,5,0)) and kernel["MatrixInstB"] == 1 and \
-         (kernel["ProblemType"]["DataType"].isHalf() or \
-          kernel["ProblemType"]["DataType"].isBFloat16() or \
-          kernel["ProblemType"]["DataType"].isInt8() or \
-          kernel["ProblemType"]["DataType"].is8bitFloat()):
-        mi_divisor = 4
-        miIssueLatency = 1
-      if (self.states.version == (9,5,0) and numBytes ==2):
-        mi_divisor = 2
-        miIssueLatency = 2
-
-      if kernel["ProblemType"]["Sparse"] or (kernel["EnableF32XdlMathOp"] and kernel["ProblemType"]["F32XdlMathOp"].isXFloat32()):
-        mi_divisor = 4
-
-      # special checking : F8 MFMA takes 2x more cycles and computes 4xK in gfx950
-      if (self.states.version == (9,5,0) and kernel["ProblemType"]["DataType"].is8bitFloat()):
-          mi_divisor = 2
-
-      self.states.miLatency = kernel["MatrixInstM"] // mi_divisor
+      from rocisa.instruction import getMFMAIssueLatency, getSMFMAIssueLatency
+      self.states.miLatency, miIssueLatency = getSMFMAIssueLatency(kernel["ProblemType"]["DataType"].toEnum(), kernel["MatrixInstM"], kernel["MatrixInstB"]) if kernel["ProblemType"]["Sparse"] else \
```

Comment:
@vin-huang  The miLatency of smfma is moved to the getSMFMAIssueLatency function.